`'/debug:fastlink'` and `'/INCREMENTAL'` aren't recognized in Windows build

Ref: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12055855519/job/33617129760?pr=2847

```bash
  LINK : warning LNK4044: unrecognized option '/debug:fastlink'; ignored
  LINK : warning LNK4044: unrecognized option '/INCREMENTAL'; ignored
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,7 +70,6 @@ else()
   set(CMAKE_EXE_LINKER_FLAGS_TRITONRELBUILDWITHASSERTS "/debug:fastlink /INCREMENTAL")
   set(CMAKE_MODULE_LINKER_FLAGS_TRITONRELBUILDWITHASSERTS "/debug:fastlink /INCREMENTAL")
   set(CMAKE_SHARED_LINKER_FLAGS_TRITONRELBUILDWITHASSERTS "/debug:fastlink /INCREMENTAL")
-  set(CMAKE_STATIC_LINKER_FLAGS_TRITONRELBUILDWITHASSERTS "/debug:fastlink /INCREMENTAL")
 endif()
 
 # Default build type
@@ -153,7 +152,7 @@ endfunction()
 if(NOT MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-covered-switch-default -fvisibility=hidden")
 else()
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX- /wd4244 /wd4624 /wd4715 /wd4530")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4244 /wd4624 /wd4715 /wd4530")
 endif()
 
 include_directories(".")

--- a/cmake/AddTritonUnitTest.cmake
+++ b/cmake/AddTritonUnitTest.cmake
@@ -28,7 +28,9 @@ function(add_triton_ut)
           gmock
           ${__LIBS})
 
-  target_compile_options(${__NAME} PRIVATE -fno-rtti)
+  if(NOT MSVC)
+    target_compile_options(${__NAME} PRIVATE -fno-rtti)
+  endif()
 
   target_compile_definitions(${__NAME} PRIVATE ${__DEFS})
 

--- a/lib/Instrumentation/CMakeLists.txt
+++ b/lib/Instrumentation/CMakeLists.txt
@@ -36,5 +36,7 @@ foreach( plugin ${GPU_INSTRUMENTATION_PASSES} )
      # This is set to -fvisibility=hidden in the top level CMake file
      # which causes the llvmGetPassPluginInfo symbol to be hidden and
      # an "entry point not found" error. Reset it just for this target
-     target_compile_options(${plugin} PRIVATE -fvisibility=default -fno-rtti)
+     if(NOT MSVC)
+         target_compile_options(${plugin} PRIVATE -fvisibility=default -fno-rtti)
+     endif()
 endforeach()

--- a/test/lib/Instrumentation/CMakeLists.txt
+++ b/test/lib/Instrumentation/CMakeLists.txt
@@ -34,5 +34,7 @@ foreach( plugin ${GPU_INSTRUMENTATION_PASSES} )
      # This is set to -fvisibility=hidden in the top level CMake file
      # which causes the llvmGetPassPluginInfo symbol to be hidden and
      # an "entry point not found" error. Reset it just for this target
-     target_compile_options(${plugin} PRIVATE -fvisibility=default)
+     if(NOT MSVC)
+         target_compile_options(${plugin} PRIVATE -fvisibility=default)
+     endif()
 endforeach()

